datahub/user/company_list/serializers.py

```diff
@@ -85,6 +85,21 @@ def validate_company(self, company):
 
         return company
 
+    def validate(self, data):
+        """
+        Raise a validation error if anything else other than allowed fields is updated.
+        """
+        if self.partial and self.instance:
+            allowed_fields = ['status']
+            fields = list(data.keys())
+            field_delta = list(set(fields) - set(allowed_fields))
+            if field_delta:
+                formated_fields = ', '.join(field_delta)
+                raise serializers.ValidationError(
+                    f'"{formated_fields}" is not allowed to be updated.',
+                )
```

Comment:
It should be possible to pass a dict to the `ValidationError`:

```suggestion
                errors = {field: default_error_messages['field_cannot_be_updated']}
                raise serializers.ValidationError(errors)
```

(You will need to define `default_error_messages` for this, for example: https://github.com/uktrade/data-hub-api/blob/a1974dae97017855cf536dc3a3fd2f248852d40d/datahub/company/serializers.py#L278-L281)